src/checkers/inference/solver/backend/encoder/ConstraintEncoderCoordinator.java

```diff
@@ -32,7 +32,7 @@
 
     public static <ConstraintEncodingT> ConstraintEncodingT dispatch(
             BinaryConstraint constraint, BinaryConstraintEncoder<ConstraintEncodingT> encoder) {
-        SlotSlotCombo combo = SlotSlotCombo.valueOf(constraint);
+        SlotSlotCombo combo = SlotSlotCombo.valueOf(constraint.getFirst(), constraint.getSecond());
```

Comment:
My idea is that we put this into `BinaryConstraint`, ie create `BinaryConstraint.getSlotSlotCombo()` which returns this value. I think its better to encapsulate the choice of the slots into the definition of the constraint itself since each instance of a Binary constraint should already know whether it is CC/CV/VC/VV.

Reply:
Well, if you think from another point of view, `dispatch(XXXConstraint)` method also should know how to dispatch each `XXXConstraint` instance.

If we have multiple places that requires this `getSlotSlotCombo()` logic, I agree with you, the `XXXConstraint` is the best place to keep this knowledge.

As currently `dispatch()` method is the only place using this logic, encapsulate this one line code to an instance method seems a bit overkill to me.

What do you think? @jyluo

Reply:
I'm good with what we have for now. Can always improve in the future (and the fact that dispatch(constraint, encoder) should always be matched pairs ... which means that in the future maybe the constraints should carry a reference to their encoder.